test/unit/spec/narrative_core/upload/stagingAreaViewer-spec.js

```diff
@@ -229,5 +235,105 @@ define ([
+
+            menuOptions.forEach(option => {
+                expect(foundOptions).toContain(option);
+            });
+
+        });
+
+        it('renders the dropdown correctly when a type is selected', async () => {
+            await stagingViewer.render();
+
+            //find the fake sra reads row specifically (via the download button, then chaining back up to the select dropdown above - since we don't have a unique ID for these select drodpowns it's the best mehtod for now)
+            let selectDropdown = $targetNode.find('[data-download="fake_sra_reads.sra"]').siblings('select');
+
+            //set the value of the dropdown
+            selectDropdown.val('sra_reads')
+                .trigger('change')
+                .trigger('select2:select');
+           
+            //check that the dropdown renders correctly
+            let select2 = $targetNode.find('[title="SRA Reads"]');
+            expect(select2).toBeDefined();
+            expect(select2.attr('title')).toContain('SRA Reads');
+            expect(select2.html()).toContain('SRA Reads');
+        });
+
+        it('should render checboxes for the file table', async () => {
+            await stagingViewer.render();
+
+            //initially the checkboxes are rendered disabled until a user selects a type
+            const tableCheckboxes = $targetNode.find('input.kb-staging-table-body__checkbox-input:disabled');
+
+            expect(tableCheckboxes.length).toBeGreaterThan(0);
+            expect(tableCheckboxes.attr('aria-label')).toContain('Select to import file checkbox: disabled until at least one data type is selected');
+
+            const headerCheckbox = $targetNode.find('#staging_table_select_all');
+            
+            expect(headerCheckbox.length).toEqual(1);
+            expect(headerCheckbox.attr('aria-label')).toContain('Select to import all files checkbox: disabled until at least one data type is selected');
+        });
+
+        it('checkboxes will be enabled when a type is selected', async () => {
+
+            await stagingViewer.render();
+
+            //find the fake sra reads one specifically
+            let selectDropdown = $targetNode.find('[data-download="fake_sra_reads.sra"]').siblings('select');
+
+            selectDropdown.val('sra_reads')
+                .trigger('change')
+                .trigger('select2:select');
+
+            //check that the table checkbox is enabled
+            const tableCheckbox = $targetNode.find('input.kb-staging-table-body__checkbox-input:enabled');
+
+            expect(tableCheckbox.length).toEqual(1);
+            expect(tableCheckbox.attr('aria-label')).toContain('Select to import file checkbox');
+
+            const headerCheckbox = $targetNode.find('#staging_table_select_all');
+            
+            //TODO: for some weird reason the header checkbox isn't showing as enabled, even though the click event fires. not sure what is going on here
```

Comment:
this is one part of the tests i couldn't quite figure out. 

when i set the dropdown value above on line 293, this triggers the select2:select event in stagingAreaViewer on 491. i am able to confirm this event gets fired, as the above check to make sure the checkbox input is enabled works. 

however for some reason the updates to the header "select all" checkbox don't get applied, even though they are in the same method call, on line 506.

the test still "works" because the element exists, but it's not really a fair test of the functionality that we are expecting (that the checkbox should be enabled). any advice is welcome